var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -119,6 +120,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         "skylake": "BDW",
         "icelake": "SKX",
         "skylake_avx512": "SKX",
+        "sophgo2042": "RISCV_SG2042",
```

Comment:
I think we should add this as an official spack architecture

Reply:
I pushed a commit to remove this, it show appear soon. Once that's the case I think this will be ready to go.

Reply:
Perhaps we should also try to use archspec in Kokkos.